src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,363 @@
+            "metadata": {
+                "location": transaction_name,
+                "title": transaction_name,
+            },
+            "nodestore_insert": segment_span.get("precise.finish_ts"),
+            "received": segment_span.get("precise.finish_ts"),
+            "request": {
+                # missing from indexed spans dataset
+            },
+            "sdk": {
+                "name": segment_sentry_tags.get("sdk.name"),
+                "version": segment_sentry_tags.get("sdk.version"),
+                # other props missing from indexed spans dataset
+            },
+            "span_grouping_config": {
+                # missing from indexed spans dataset
+            },
+            "entries": [
+                {
+                    "type": "spans",
+                    "data": _span_data_to_event_spans(spans_data),
+                }
+            ],
+            "start_timestamp": segment_span.get("precise.start_ts"),
+            "timestamp": segment_span.get("precise.finish_ts"),
+            "title": transaction_name,
+            "transaction": transaction_name,
+            "transaction_info": {
+                # missing from indexed spans dataset
+            },
+            "type": "transaction",
+            "user": {
+                # missing from indexed spans dataset
+            },
+            "version": 5,
+            "projectSlug": project_slug,
+        }
+
+        average_columns = request.GET.getlist("averageColumn", [])
+        if (
+            all(col in VALID_AVERAGE_COLUMNS for col in average_columns)
+            and len(average_columns) > 0
+        ):
+            _add_comparison_to_event(data, organization.id, project, average_columns)
+
+        return Response(data)
+
+
+def _query_all_spans_in_transaction(organization, project, transaction_id, start, end):
+    params: ParamsType = {
+        "start": start - timedelta(seconds=1),
+        "end": end + timedelta(seconds=1),
+        "project_id": [project.id],
+        "project_objects": [project],
+        "organization_id": organization.id,
+    }
+
+    # Look up the spans for this transaction
+    query = SpansIndexedQueryBuilder(
+        Dataset.SpansIndexed,
+        params,
+        query=f"transaction_id:{transaction_id}",
+        selected_columns=[
+            "transaction_id",
+            "transaction_op",
+            "trace_id",
+            "span_id",
+            "profile_id",
+            "parent_span_id",
+            "segment_id",
+            "is_segment",
+            "segment_name",
+            "precise.start_ts",
+            "precise.finish_ts",
+            "exclusive_time",
+            "op",
+            "group",
+            "span_status",
+            "span_kind",
+            "description",
+            "status",
+            "platform",
+            "user",
+            "measurements.key",
+            "measurements.value",
+        ],
+        orderby=["precise.start_ts", "id"],
```

Comment:
should we add `is_segment` as the first orderby here so its guaranteed to show up within the 10k?